runelite-client/src/main/java/net/runelite/client/ui/overlay/OverlayRenderer.java

```diff
@@ -493,6 +493,17 @@ private void safeRender(Overlay overlay, Graphics2D graphics, Point point)
 	{
 		final Graphics2D subGraphics = (Graphics2D) graphics.create();
 		subGraphics.translate(point.x, point.y);
+		if (overlay.getLayer() == OverlayLayer.ABOVE_SCENE)
```

Comment:
Extend this to every layer lower than ABOVE_SCENE

Reply:
So it should be for ABOVE_SCENE and UNDER_WIDGETS?

Reply:
Yeah, in fixed mode anything that is on that layers should be visible only in viewport as everywhere else are game widgets (well, except minimap).

Reply:
UNDER_WIDGETS is used by overlays that can be moved outside the viewport so I think we should just clip it for ABOVE_SCENE

Reply:
Their layer is changed to ABOVE_SCENE if that happens before it will reach this method.